### PR TITLE
Sqlite everywhere

### PR DESCRIPTION
**Pull request by sdtaylor**
some much needed cleanup. I made a single database function which handles opening and closing the database and switched all data functions in forecast-bbs-core.R over to it. Let me know what you think and if it will be adequate for everything we want to do.  the other data scripts can be switched over pretty easily once we're sure this is working fine.

Also did some roxygen import updates as required. 

#38

### CODE REVIEW

File: R/forecast-bbs-core.R [hunk 1]
```diff
@@ -82,40 +127,50 @@ get_species_data = function() {
...
+#' data. Install it via ecodataretriever if needed. 
+#' 
 #' @export
+#' @importFrom dplyr "%>%" group_by
 get_bbs_data <- function(){
-  # Get the BBS data
 
   data_path <- paste('./data/', 'bbs', '_data.csv', sep="")
   if (file.exists(data_path)){
     return(read.csv(data_path))
```
Comment (davharris) [inline R/forecast-bbs-core.R]: `readr::read_csv` is substantially faster here.

**Comment by davharris** [comment]
I've skimmed it and it seems great so far.  Nice work.  I'll try to run it on my machine over the weekend.

**Comment by davharris** [comment]
Namespace problem 😢

I think the fix is to add `#' @importFrom sp coordinates` between lines 54 and 55 of `get_prism_data.R`

I think the warnings at the bottom are unrelated. [Apparently, connections get left open when you `dbFetch` only some of the rows](https://stackoverflow.com/questions/4084028/how-to-close-resultset-in-rmysql). Apparently you can close it with `dbClearResult` or you might be able to use `dbGetQuery` to avoid the issue altogether.

```
 Error in data.frame(site_id = locations$site_id, coordinates(locations),  : 
  could not find function "coordinates" 
18.
data.frame(site_id = locations$site_id, coordinates(locations), 
    extracted) at get_prism_data.R#84
17.
get_prism_data() at get_prism_data.R#128
16.
process_bioclim_data() at get_prism_data.R#192
15.
get_bioclim_data() at forecast-bbs-core.R#189
14.
get_env_data() at forecast-bbs-core.R#277
13.
add_env_data(.) 
12.
function_list[[i]](value) 
11.
freduce(value, `_function_list`) 
10.
`_fseq`(`_lhs`) 
9.
eval(expr, envir, enclos) 
8.
eval(quote(`_fseq`(`_lhs`)), env, env) 
7.
withVisible(eval(quote(`_fseq`(`_lhs`)), env, env)) 
6.
richness_data %>% add_env_data() %>% filter(!is.na(bio1), !is.na(ndvi_sum), 
    !is.na(elevs)) at forecast-bbs-core.R#265
5.
get_richness_ts_env_data(start_yr, end_yr, min_num_yrs) at .active-rstudio-document#5
4.
eval(expr, envir, enclos) 
3.
eval(ei, envir) 
2.
withVisible(eval(ei, envir)) 
1.
source("~/.active-rstudio-document", echo = TRUE) 
In addition: Warning messages:
1: In .local(conn, ...) : Closing open result set
2: In .local(conn, ...) : Closing open result set
3: In .local(conn, ...) : Closing open result set
```

**Comment by ethanwhite** [comment]
I'm a bit overrun with the start of the semester, so @davharris, once you think it's ready, go ahead and merge.

**Comment by davharris** [comment]
Reading through the changes, it looks good.  Will start re-running it now, and will pull if I don't see any problems.  Thanks again for tackling this.

**Comment by sdtaylor** [comment]
make sure to grab this latest commit. I took out the DBI stuff which was causing issues.

**Comment by davharris** [comment]
@sdtaylor yeah, I got errors. Running the new version now...

**Comment by davharris** [comment]
`Error in sqliteSendQuery(conn, statement) : 
  error in statement: near ";": syntax error`

From the `traceback()`, it looks like the error is near the `sql_query` in `collect(tbl(con, sql(sql_query))) at forecast-bbs-core.R#37`

**Comment by sdtaylor** [comment]
I'm slowly learning to run new code from the very beginning *every* time.

**Comment by davharris** [comment]
I get an error if I try to run it without a `data` folder.  Can you create one just before `con <- src_sqlite(db, create=TRUE)`? Like, 

`if(!dir.exists("data")){dir.create("data")}`

Also, can you pull in the latest changes to master to make sure we don't get a warning about "This branch has conflicts that must be resolved?"  Once you do, I'll test it again.

Thanks again for all your hard work on this.  I'm sure it's been a huge pain.

**Comment by sdtaylor** [comment]
I ran this latest commit on serenity and it worked from an empty data folder all the way thru to richness+environmental data, so hopefully it works for you too.

**Comment by davharris** [comment]
no errors, but I get

```
Warning message:
Only first 100,000 results retrieved. Use n = Inf to retrieve all.
```

Try adding `n = inf` to the `collect` call on line 39.

**Comment by sdtaylor** [comment]
I've yet to see that warning. what command are you calling?

**Comment by ethanwhite** [comment]
Probably a difference in `dplyr` versions. I believe this only getting the first 100K results is a relatively new behavior.

**Comment by sdtaylor** [comment]
thanks @ethanwhite, updating dplyr to 0.5.0 brought up the warning

**Comment by davharris** [comment]
Looks good to me.  I'm going to merge now and then we can quibble about whether things match exactly across machines on #64 or a new commit.

Thanks again!

**Pull request merged**